cmake/AddGncDocTargets.cmake

```diff
@@ -22,21 +22,42 @@ function (add_gnc_doc_targets lang entities figures)
         add_dependencies(${lang}-check "${lang}-${targetbase}-check")
     endif()
 
+    set(xslargs "${ARGN}")
```

Comment:
Your extra code does what you intended. It is however missing a final bit: the newly added xsl documents should somehow be added to the list of files to distribute. Otherwise the release tarball would fail to build (and you can probably verify this by running make/ninja dist-check, which I haven't ATM).

There are a few ways to accomplish this:
One would be to add the files in the `add_to_dist` call in `zh/CMakeLists.txt`. That's a simple and straightforward solution. It would mean however users would have to remember adding this for each new xsl file. I had written the build system such that anything in the entities or figures lists would automatically be added to the dist files. For consistency I would prefer to have a similar solution for xsl files.

Proposal:
Extend the function signature of `add_gnc_doc_targets` to also have an explicit `xsl_files` parameter and use that instead of `${ARGN}`. Your parsing code here would remain the same for the normal build scripts.
But for the `add_to_dist` further down in this CMakeLists.txt file, you reparse to extract all the xml files to add for distribution.

In the calling CMakeLists.txt files (like `zh/guide/CMakeLists.txt`), formally set a variable called `xsl_files` and pass that to the call to `add_gnc_doc_targets`. So you'd get

```
set(xsl_files
     "xsl:pdf" "${CMAKE_CURRENT_SOURCE_DIR}/../xsl/zh-fo.xsl"
     "xsl:chm" "${CMAKE_CURRENT_SOURCE_DIR}/../xsl/zh-chm.xsl")

add_gnc_doc_targets(${lang} "${entities}" "${figures}" "${xsl_files}")
```

While that's functionally equivalent to what you did, it makes it more clear the function accepts a list of xsl files (with type spec) as extra parameter. That would make it a good example other languages could follow in the future.

Note: you have added several new files, but most of the others will be picked up by the build system code already, albeit after an explicit rerun of cmake (and perhaps even one that's not using the cached variables). From what I can see from direct review, only the two xsl files were missing.

Reply:
@xuxinhang Do you need some help for this?

Reply:
The common xsl files are added to dist here:
https://github.com/Gnucash/gnucash-docs/blob/98a176d508852c9f0295c3fc19141bf6dd686a11/CMakeLists.txt#L186

The fact they are not explicitly listed could be considered a design flaw (partly by my own doing :sad: ). I think all files that are handwritten as part of the gnucash-docs should be individually listed before adding them to dist. This is different for the full stylesheet that was imported (and should ideally not be altered by us).

Reply:
@gjanssens can you add a section about adding xsl files to https://wiki.gnucash.org/wiki/Documentation_Improvement#Add_Extra_Files?
OTOH I assume adding fonts and xslts is more a task for https://wiki.gnucash.org/wiki/Documentation_Translation?

Reply:
I don't know what you want me to add. With the changes @xuxinhang made to the build system, adding an xslt file is very similar to adding a figure. I don't immediately see that documented either and am tempted to just leave this at "learn by example".